### PR TITLE
[Pager] Deprecating accompanist pager in favour of androidx pager.

### LINKED ISSUE #1463: [Pager] Add migration guide to new Pager API in androidx.compose.foundation

### PR DESCRIPTION
**Pull request by riggaroo**
Deprecating `HorizontalPager` and `VerticalPager` in accompanist.

Fixes #1463

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,3 +29,13 @@ All submissions, including submissions by project members, require review. We
 use GitHub pull requests for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using pull requests.
+
+## API Changes
+
+If you are changing any public APIs, you need to run `./gradlew :pager:metalavaGenerateSignature` which will
```
Comment (bentrengrove) [inline CONTRIBUTING.md:35]: Sorry this wasn't documented anywhere. 

You can just do `./gradlew metalavaGenerateSignature` same with spotless

File: README.md [hunk 2]
```diff
@@ -46,8 +46,8 @@ A library that enables the reuse of [MDC-Android][mdc] Material 2 XML themes, fo
 ### 🎨 [Material 3 Theme Adapter](./themeadapter-material3/)
 A library that enables the reuse of [MDC-Android][mdc] Material 3 XML themes, for theming in Jetpack Compose.
 
-### 📖 [Pager](./pager/) (Soon to be deprecated - Upstreamed in Compose 1.4)
-A library that provides utilities for building paginated layouts in Jetpack Compose, similar to Android's [ViewPager][viewpager].
+### 📖 [Pager](./pager/) (Deprecated)
```
Comment (bentrengrove) [inline README.md:49]: Please move to bottom of the readme with the other deprecated libraries

File: docs/pager.md [hunk 3]
```diff
@@ -5,8 +5,44 @@
 A library which provides paging layouts for Jetpack Compose. If you've used Android's [`ViewPager`](https://developer.android.com/reference/kotlin/androidx/viewpager/widget/ViewPager) before, it has similar properties.
 
 !!! warning
-    The pager layouts are currently experimental and the APIs could change at any time.
-    All of the APIs are marked with the `@ExperimentalPagerApi` annotation.
+**This library is deprecated, with official pager support in `androidx.compose.foundation.pager` ** 
+The migration guide and original documentation is below the migration guide.
```
Comment (bentrengrove) [inline docs/pager.md:9]:
```suggestion
The original documentation is below the migration guide.
```

**Review by Levi-Moreira — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (riggaroo) [reply CONTRIBUTING.md]: Done

File: README.md (hunk 2 above)
Comment (riggaroo) [reply README.md]: Done

**Pull request merged**